analyzer.go

```diff
@@ -52,205 +65,236 @@ type Analyzer struct {
 	template string
 }
 
-func (a *Analyzer) processPerTargetValues(target *Target) error {
-	a.values["mod-year"] = a.values["year"]
-	a.values["mod-year-range"] = a.values["year-range"]
-	if t, err := target.ModTime(); err == nil {
-		a.values["mod-year"] = &ConstValue{RawValue: fmt.Sprint(t.Year())}
-		a.values["mod-year-range"] = &RegexpValue{RawValue: `((20\d\d\-{{mod-year}})|({{mod-year}}))`}
+func New(opts ...Option) *Analyzer {
+	var a Analyzer
+	for _, opt := range opts {
+		opt.apply(&a)
 	}
+	return &a
+}
 
-	for _, v := range a.values {
-		if err := v.Calculate(a.values); err != nil {
-			return err
-		}
+type Result struct {
+	Err        error
+	Fix        string
+	End, Start token.Pos
+}
+
+func (a *Analyzer) skipCodeGen(file *ast.File) ([]*ast.CommentGroup, []*ast.Comment) {
+	var comments = file.Comments
+	var list []*ast.Comment
+	if len(comments) > 0 {
+		list = comments[0].List
 	}
-	return nil
+	// for len(comments) > 0 && strings.Contains(comments[0].Text(), "DO NOT EDIT") {
+	// 	comments = comments[1:]
+	// 	if len(comments) > 0 {
+	// 		list = comments[0].List
+	// 		if len(list) > 0 && strings.HasSuffix(list[0].Text, "//line:") {
+	// 			list = list[1:]
+	// 		}
+	// 	}
+	// }
+	return comments, list
 }
 
-func (a *Analyzer) Analyze(target *Target) (i Issue) {
+func (a *Analyzer) Analyze(t *Target) (result *Result) {
```

Comment:
I think the signature should be:
```go
func (a *Analyzer) Analyze(path string, File *ast.File) error {
```

and the `Diagnostics` reported inside this method.

Reply:
Good catch, fixed.